packages/syft/src/syft/service/code/user_code.py

```diff
@@ -362,15 +414,76 @@ def user(self) -> UserView | SyftError:
             )
         return api.services.user.get_by_verify_key(self.user_verify_key)
 
+    def _status_from_output_history(
+        self, context: AuthedServiceContext | None = None
+    ) -> UserCodeStatusCollection | SyftError:
+        if context is None:
+            # Clientside
+            api = self._get_api()
+            if isinstance(api, SyftError):
+                return api
+            node_identity = NodeIdentity.from_api(api)
+            is_approved = api.output.has_output_read_permissions(
+                self.id, self.user_verify_key
+            )
+        else:
+            # Serverside
+            node_identity = NodeIdentity.from_node(context.node)
+            output_service = context.node.get_service("outputservice")
+            is_approved = output_service.has_output_read_permissions(
+                context, self.id, self.user_verify_key
+            )
+
+        if isinstance(is_approved, SyftError):
+            return is_approved
+        is_denied = self.l0_deny_reason is not None
+
+        if is_denied:
+            message = self.l0_deny_reason
+            status = (UserCodeStatus.DENIED, message)
+        elif is_approved:
+            status = (UserCodeStatus.APPROVED, "")
+        else:
+            status = (UserCodeStatus.PENDING, "")
+        status_dict = {node_identity: status}
+
+        return UserCodeStatusCollection(
+            status_dict=status_dict,
+            user_code_link=LinkedObject.from_obj(self),
+        )
+
     @property
     def status(self) -> UserCodeStatusCollection | SyftError:
         # Clientside only
+
+        if self.is_low_side:
+            if self.status_link is not None:
```

Comment:
this should be about the deployment type. We can make a separate ticket to include a deployment setting